### PR TITLE
Add staffs screen

### LINKED ISSUE #190: Add staffs screen with design
**Idea Description**
Please add staffs screen and make it possible to transition from the about screen.

There is no figma design for this screen, but it is assumed to be similar to the contributors screen.

**Reference images and links**
[Figma](https://www.figma.com/file/NcSMs6dMsD88d4wOY0g3rK/Conference-App?node-id=448%3A2009)
about-screen

### PR DESCRIPTION
**Pull request by t-nonomura**
## Issue
- close #190 

## Overview (Required)
- Implement the transition from the about screen to the staff screen.
  - It has almost the same structure as the contributors screen.
  - Add feature-staff module.
    - staff screen and ui test
  - Add repository, api and response model to the core-data module.

## Links
- Figma
  - https://www.figma.com/file/NcSMs6dMsD88d4wOY0g3rK/Conference-App

## Screenshot
Before | After (picture) | After (movie) 
:--: | :--: | :--:
none |  | 

|FYI : contributors screen|
|:--:|
||

### CODE REVIEW

**Comment by RyuNen344** [comment]
Thanks for contribution!!
Could you resolve conflicts??

**Comment by t-nonomura** [comment]
@RyuNen344 
Thanks for pointing out, I resolved the conflict:pray:

**Comment by t-yusaku** [comment]
@t-nonomura 
Could you run `./gradlew spotlessApply` ? 🙏
It resolves the lint warnings.

**Comment by t-nonomura** [comment]
@t-yusaku 
I'm checking the error details, but `./gradlew spotlessApply` could not be executed due to the following error.
```
An exception occurred applying plugin request [id: 'droidkaigi.primitive.dependencygraph']
> Could not find implementation class 'io.github.droidkaigi.confsched2022.primitive.DependencyGraphPlugin' for plugin 'droidkaigi.primitive.dependencygraph' specified in jar:file:/Users/username/.gradle/caches/jars-9/89627793f7570c1589b13b2439a6f06c/build-logic.jar!/META-INF/gradle-plugins/droidkaigi.primitive.dependencygraph.properties.

```

I found the problem from the build log, so I fixed it manually for the time being.
[914e7c1](https://github.com/DroidKaigi/conference-app-2022/pull/331/commits/914e7c1fd8a508cc030659ee3eabe8aa19ffa98e)

**Comment by takahirom** [comment]
Sorry. Can you run spotlessKotlinApply?

**Comment by t-nonomura** [comment]
@takahirom 
Sorry, `./gradlew spotlessApply` succeeded after deleting all gradle caches. (`rm -rf  ~/.gradle/caches/`)
Also the same for spotlessKotlinApply:bow:

**Comment by takahirom** [comment]
Sorry for bothering you. Can you take a look at the conflict?

File: feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/StaffUiModel.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package io.github.droidkaigi.confsched2022.feature.staff
+
+import io.github.droidkaigi.confsched2022.model.Staff
+import io.github.droidkaigi.confsched2022.ui.Result
+import kotlinx.collections.immutable.PersistentList
+
+data class StaffUiModel(
+    val staffState: StaffState,
+) {
+    sealed interface StaffState {
```
Comment (kittinunf) [inline feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/StaffUiModel.kt:10]: @t-nonomura Do you think you can try to use UiLoadState introduced in #287? We could make the UiModel even simpler to use.

File: feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            CircularProgressIndicator()
+            return@KaigiScaffold
+        }
+        val staff = uiModel.staffState.staff
+
+        LazyColumn(
+            modifier = modifier.fillMaxWidth()
+        ) {
+            items(items = staff, key = { it.id }) { staff ->
+                Row(
```
Comment (kittinunf) [inline feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt:74]: This item is the same as the contributor [row](https://github.com/DroidKaigi/conference-app-2022/blob/main/feature-contributors/src/main/java/io/github/droidkaigi/confsched2022/feature/contributors/Contributors.kt#L87-L114). 

Do you think we could share the UiComponent?

File: feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/StaffUiModel.kt (hunk 1 above)
Comment (t-nonomura) [reply feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/StaffUiModel.kt]: I fixed it.
d53e563

Your code is very concise, thank you !

File: feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt (hunk 2 above)
Comment (t-nonomura) [reply feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt]: @kittinunf 
Thanks for the great point out. 
I have a consultation.

I think this refactoring should add modules to put common UI components.
If I respond with this Pull Request, the change will be large, so is it okay to create a new issue?
Like a  "Add a module for placing common UI parts, such as the `common-ui-component` module"

I'll wait and see and if no one is assigned, I'll raise my hand.

File: feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/StaffUiModel.kt (hunk 1 above)
Comment (kittinunf) [reply feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/StaffUiModel.kt]: 👍

File: feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt (hunk 2 above)
Comment (kittinunf) [reply feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt]: I feel like core-ui might be a good candidate for this, it should be a foundation module for all of the feature modules. What do you think?

cc: @takahirom

Comment (takahirom) [reply feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt]: How about having it here?
https://github.com/DroidKaigi/conference-app-2022/tree/main/core-designsystem/src/androidMain/kotlin/io/github/droidkaigi/confsched2022/designsystem/components

**Comment by takahirom** [comment]
Let me see this tomorrow.

File: feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt (hunk 2 above)
Comment (t-nonomura) [reply feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt]: Thanks, I had overlooked core-designsystem module.
looks good.

Sorry for the delay, I’ll fix it at a convenient time today.

**Comment by takahirom** [comment]
Please resolve conflict and move the function to core-designsystem 🙏 
You don't have to move to core-designsystem, if you don't have time.

File: feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt (hunk 2 above)
Comment (t-nonomura) [reply feature-staff/src/main/java/io/github/droidkaigi/confsched2022/feature/staff/Staff.kt]: @takahirom 
I created the rows used in the staff list and contributors as a common component.
[89af427](https://github.com/DroidKaigi/conference-app-2022/pull/331/commits/89af427d922ffeeff0e0810091218c47ecc1d2e5)

File: app-android/src/main/java/io/github/droidkaigi/confsched2022/KaigiApp.kt [hunk 3]
```diff
@@ -103,27 +105,33 @@ fun KaigiApp(
...
+                        onBackIconClick = kaigiAppScaffoldState::onBackIconClick,
+                        onSearchIconClick = kaigiAppScaffoldState::onSearchClick,
+                        onTimetableClick = kaigiAppScaffoldState::onTimeTableClick,
+                        onNavigateFloorMapClick = kaigiAppScaffoldState::onNavigateFloorMapClick,
                     )
                     contributorsNavGraph(
-                        showNavigationIcon,
-                        kaigiAppScaffoldState::onNavigationClick,
+                        showNavigationIcon = showNavigationIcon,
+                        onNavigationIconClick = kaigiAppScaffoldState::onNavigationClick,
```
Comment (takahirom) [inline app-android/src/main/java/io/github/droidkaigi/confsched2022/KaigiApp.kt:117]: 👍

**Review by takahirom — COMMENTED** [review]
LGTM! Thank you for creating the foundation of DroidKaigi!

**Pull request merged**